compiler-lsp/analyzer/src/document_highlight.rs

```
@@ -334,17 +791,8 @@ fn push_document_highlight(
-        highlights.push(highlight);
-    }
-}
-
-fn document_highlights(
-    content: &str,
-    encoding: PositionEncoding,
-    ranges: impl IntoIterator<Item = Utf8Range>,
-) -> Vec<DocumentHighlight> {
-    ranges.into_iter().filter_map(|range| document_highlight(content, encoding, range)).collect()
+    let highlight = range.and_then(|range| document_highlight(content, encoding, range));
+    highlights.extend(highlight.into_iter());
```

Clippy (1.96.0)

[warning] explicit call to `.into_iter()` in function argument accepting `IntoIterator`

(warning)